activity/src/main/java/com/amazonaws/stepfunctions/cloudformation/activity/DeleteHandler.java

```diff
@@ -1,49 +1,136 @@
 package com.amazonaws.stepfunctions.cloudformation.activity;
 
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.retry.PredefinedRetryPolicies;
 import com.amazonaws.services.stepfunctions.AWSStepFunctions;
+import com.amazonaws.services.stepfunctions.AWSStepFunctionsClientBuilder;
 import com.amazonaws.services.stepfunctions.model.DeleteActivityRequest;
 import com.amazonaws.services.stepfunctions.model.DescribeActivityRequest;
+import com.google.common.util.concurrent.Uninterruptibles;
 import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
 import software.amazon.cloudformation.proxy.Logger;
 import software.amazon.cloudformation.proxy.OperationStatus;
 import software.amazon.cloudformation.proxy.ProgressEvent;
 import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
 
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
 public class DeleteHandler extends ResourceHandler {
 
+    private static final int RETRY_COUNTER = 3;
+    private static final int INITIAL_RETRY_DELAY_SECONDS = 10;
+    private static final int MAX_RETRY_DELAY_SECONDS = 30;
+    private static final int STABILIZATION_CALL_COUNT = 10;
+    private static final Random random = new Random();
+
     @Override
     public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
 
         logger.log("INFO Activity DeleteHandler with clientRequestToken: " + request.getClientRequestToken());
 
         final ResourceModel model = request.getDesiredResourceState();
 
         try {
-            verifyActivityArnIsPresent(model.getArn());
+            if (callbackContext == null || !callbackContext.isActivityDeletionStarted()) {
+                return initiateDeleteActivity(proxy, model, logger);
+            } else {
+                int currentRetryCount = callbackContext.getRetryCount();
+                if (currentRetryCount <= RETRY_COUNTER) {
+                    callbackContext.setRetryCount(currentRetryCount + 1);
+                    return stabilizeDeleteActivity(proxy, model, callbackContext.getRetryCount(), logger);
```

Comment:
Might be unnecessary to call `callbackContext.getRetryCount` when we already set it a line above

```suggestion
                    return stabilizeDeleteActivity(proxy, model, currentRetryCount + 1, logger);
```